src/main/java/redis/clients/jedis/Connection.java

```diff
@@ -88,24 +89,34 @@ protected Connection sendCommand(final ProtocolCommand cmd, final String... args
     return sendCommand(cmd, bargs);
   }
 
-  protected Connection sendCommand(final ProtocolCommand cmd, final byte[]... args) {
-    try {
-      connect();
-      Protocol.sendCommand(outputStream, cmd, args);
-      return this;
-    } catch (JedisConnectionException ex) {
-      // Any other exceptions related to connection?
-      broken = true;
-      throw ex;
-    }
+  protected Connection sendCommand(final ProtocolCommand cmd) {
+    return sendCommand(cmd, EMPTY_ARGS);
   }
 
-  protected Connection sendCommand(final ProtocolCommand cmd) {
+  protected Connection sendCommand(final ProtocolCommand cmd, final byte[]... args) {
     try {
       connect();
-      Protocol.sendCommand(outputStream, cmd, new byte[0][]);
+      Protocol.sendCommand(outputStream, cmd, args);
       return this;
     } catch (JedisConnectionException ex) {
+      /*
+       * When client send request which formed by invalid protocol, Redis send back error message
+       * before close connection. We try to read it to provide reason of failure.
+       */
+      try {
+        // Set soTimeout 100ms to avoid long blocking
+        socket.setSoTimeout(100);
+        String errorMessage = Protocol.readErrorLineIfPossible(inputStream);
```

Comment:
Do we really need the `readErrorLineIfPossible` method? Can't we just call `Protocol.read(inputStream)` and catch the JedisDataException and convert it into a JedisConnectionException?. I believe the code in the `readErrorLineIfPossible` is pretty much like what `Protoco.read` does when an error is returned.